include/core/shape.h

```diff
@@ -378,6 +378,178 @@ class Sphere : public Shape<dim>
+    const unsigned int /*component = 0*/) override;
+
+  /**
+   * @brief Return a pointer to a copy of the Shape
+   */
+  std::shared_ptr<Shape<dim>>
+  static_copy() const override;
+
+  /**
+   * @brief Return the analytical gradient of the distance
+   * @param evaluation_point The point at which the function will be evaluated
+   * @param component This parameter is not used, but it is necessary because Shapes inherit from the Function class of deal.II.
+   */
+  Tensor<1, dim>
+  gradient(const Point<dim> & evaluation_point,
+           const unsigned int component = 0) const override;
+
+  /**
+   * @brief Return the gradient of the distance function
+   * @param evaluation_point The point at which the function will be evaluated
+   * @param cell The cell that is likely to contain the evaluation point
+   * @param component Not applicable
+   */
+  Tensor<1, dim>
+  gradient_with_cell_guess(
+    const Point<dim> &                                   evaluation_point,
+    const typename DoFHandler<dim>::active_cell_iterator cell,
+    const unsigned int component = 0) override;
+
+  /**
+   * @brief
+   * Sets the closest_point parameter to be the point on the surface of the
+   * shape which has the minimal distance from the given point p. Since this
+   * function is at the core of the calculations for superquadrics, it is
+   * redefined here.
+   *
+   * @param p The point at which the evaluation is performed
+   * @param closest_point The reference to the closest point. This point will be modified by the function.
+   * @param cell_guess A guess of the cell containing the evaluation point, which
+   * is useful to reduce computation time
+   */
+  void
+  closest_surface_point(
+    const Point<dim> &                                    p,
+    Point<dim> &                                          closest_point,
+    const typename DoFHandler<dim>::active_cell_iterator &cell_guess) override;
+  void
+  closest_surface_point(const Point<dim> &p,
+                        Point<dim> &      closest_point) const override;
+
+
+  /**
+   * @brief
+   * Clear the cache of the shape
+   */
+  virtual void
+  clear_cache() override;
+
+  /**
+   * @brief Return the sign of the parameter. To be removed once PR#794 is merged.
+   * @param prm parameter
+   */
+  inline double
+  sign(const double prm) const
+  {
+    if (prm > 0)
+      return 1;
+    else if (prm < 0)
+      return -1;
+    else
+      return 0;
+  }
+
+  /**
+   * @brief Computes the value of the superquadric from its equation
+   * @param centered_point point at which we make the evaluation, in the shape referential
+   */
+  inline double
+  superquadric(const Point<dim> &centered_point) const
+  {
+    return pow(abs(centered_point[0] / half_lengths[0]), exponents[0]) +
+           pow(abs(centered_point[1] / half_lengths[1]), exponents[1]) +
+           pow(abs(centered_point[2] / half_lengths[2]), exponents[2]) - 1;
```

Comment:
```suggestion
           pow(abs(centered_point[2] / half_lengths[2]), exponents[2]) - 1.0;
```

Reply:
What happens if one of the half_lenghts array element is equal to zero ? Maybe you could throw an error ?